src/dmn/trn_agent_xdp_usr.c

```diff
@@ -251,12 +251,51 @@ int trn_agent_delete_endpoint(struct agent_user_metadata_t *umd,
 	return 0;
 }
 
+int trn_agent_update_packet_metadata(struct agent_user_metadata_t *umd,
```

Comment:
Why are you updating from the agent side?

Reply:
I see there is update_endpoint for both agent side and transit side.